spec/lib/definition/types/keys_spec.rb

```diff
@@ -189,4 +189,36 @@
       end
     end
   end
+
+  describe ".dup" do
+    it "dups definitions and defaults" do
+      original = described_class.new("person",
```

Comment:
do we need that? In which situation would you dup a definition itself? Also I think the other definition typess do not support this as well